atomic_reactor/plugins/pre_add_yum_repo_by_url.py

```diff
@@ -52,11 +52,25 @@ def run(self):
         if self.repourls:
             for repourl in self.repourls:
                 yumrepo = YumRepo(repourl)
-                yumrepo.fetch()
-                self.log.info("fetched repo from '%s'", yumrepo.repourl)
+                self.log.info("fetching yum repo from '%s'", yumrepo.repourl)
+                try:
+                    yumrepo.fetch()
+                except Exception as e:
+                    self.log.exception(
+                        "Exception raised during yum repo fetching"
+                    )
+                    raise RuntimeError(
+                        "Failed to fetch yum repo {repo}: {exc}".format(
+                            repo=yumrepo.repourl, exc=e
+                        ))
+                else:
+                    self.log.info("fetched yum repo from '%s'", yumrepo.repourl)
+
                 if self.inject_proxy:
                     if yumrepo.is_valid():
                         yumrepo.set_proxy_for_all_repos(self.inject_proxy)
                 self.workflow.files[yumrepo.dst_filename] = yumrepo.content
-                self.log.debug("saving repo '%s', length %d", yumrepo.dst_filename,
-                               len(yumrepo.content))
+                self.log.debug(
```

Comment:
Please don't churn files unnecessarily.

Reply:
I updated message there, and it is longer than 100 chars

Reply:
sorry my bad, I looked on wrong line, will fix this